File: bilibili-recommend-ui/app/src/main/java/com/bilibilirecommend/base/BaseActivity.java
package com.bilibilirecommend.base;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

/**
 * by y on 2016/9/13
 */
public abstract class BaseActivity extends AppCompatActivity implements BaseFragment.BackHandledInterface {

    public BaseFragment baseFragment;
    public Bundle mBundle = null;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(getLayoutId());
        mBundle = getIntent().getExtras();
        initById();
        initCreate(savedInstanceState);
    }


    @Override
    public void setSelectedFragment(BaseFragment selectedFragment) {
        this.baseFragment = selectedFragment;
    }

    protected abstract void initCreate(Bundle savedInstanceState);

    protected abstract void initById();

    protected abstract int getLayoutId();

    @Override
    public void onBackPressed() {
        if (baseFragment == null || !baseFragment.onBackPressed()) {
            if (getSupportFragmentManager().getBackStackEntryCount() == 0) {
                super.onBackPressed();
            } else {
                getSupportFragmentManager().popBackStack();
            }
        }
    }

}

File: BilibiliRecommendUi/app/src/main/java/com/bilibilirecommend/base/BaseActivity.kt
package com.bilibilirecommend.base

import android.os.Bundle
import android.support.v7.app.AppCompatActivity

abstract class BaseActivity : AppCompatActivity(), BaseFragment.BackHandledInterface {

    var baseFragment: BaseFragment? = null
    var mBundle: Bundle? = null

    protected abstract val layoutId: Int

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(layoutId)
        mBundle = intent.extras
        initById()
        initCreate(savedInstanceState)
    }


    override fun setSelectedFragment(selectedFragment: BaseFragment) {
        this.baseFragment = selectedFragment
    }

    protected abstract fun initCreate(savedInstanceState: Bundle?)

    protected abstract fun initById()

    override fun onBackPressed() {
        if (baseFragment == null || baseFragment?.onBackPressed() != true) {
            if (supportFragmentManager.backStackEntryCount == 0) {
                super.onBackPressed()
            } else {
                supportFragmentManager.popBackStack()
            }
        }
    }

}
